src/umls/impl/translator.hpp

```diff
@@ -0,0 +1,153 @@
+                        _Load_integer<uint32_t>(_Buf.get() + _Off + _Catalog_name_size)
+                    });
+                }
+
+                return true;
+            }
+
+        private:
+            file_stream& _Mystream;
+        };
+
+        inline const path& _Get_settings_file_path() {
+            // concatenate the current path with the settings file name to obtain its absolute path
+            static const path& _Path = ::mjx::current_path() / L"settings.uts";
+            return _Path;
+        }
+
+        class _Translator_init_lcids_iterator { // forward iterator for translator initialization LCIDs
+        public:
+            using value_type        = uint32_t;
+            using difference_type   = ptrdiff_t;
+            using pointer           = const value_type*;
+            using reference         = const value_type&;
+            using iterator_category = ::std::forward_iterator_tag;
+
+            explicit _Translator_init_lcids_iterator(const translator_settings& _Settings) noexcept
+                : _Myelems{0}, _Mycount(0), _Myoff(0) {
+                _Init(_Settings);
+            }
+
+            ~_Translator_init_lcids_iterator() noexcept {}
+
+            _Translator_init_lcids_iterator()                                                  = delete;
+            _Translator_init_lcids_iterator(const _Translator_init_lcids_iterator&)            = delete;
+            _Translator_init_lcids_iterator& operator=(const _Translator_init_lcids_iterator&) = delete;
+
+            reference operator*() const noexcept {
+                return _Myelems[_Myoff];
+            }
+
+            _Translator_init_lcids_iterator& operator++() noexcept {
+                ++_Myoff;
+                return *this;
+            }
+
+            pointer begin() const noexcept {
+                return _Myelems;
+            }
+
+            pointer end() const noexcept {
+                return _Myelems + _Mycount;
+            }
+
+        private:
+            static constexpr size_t _Max_elem_count = 4;
+            
+            void _Insert_lcid_if_unique(const uint32_t _Lcid) noexcept {
+                // insert _Lcid to _Myelems if it is not already present, otherwise ignore it
+                for (size_t _Idx = 0; _Idx < _Mycount; ++_Idx) {
+                    if (_Myelems[_Idx] == _Lcid) { // same LCID found, break
+                        return;
+                    }
+                }
+
+                _Myelems[_Mycount++] = _Lcid;
+            }
+
+            void _Init(const translator_settings& _Settings) noexcept {
+                // Note: The order of LCIDs is fixed and defined as follows: user-preferred, user-default,
+                //       system-preferred, and system-default. To prevent multiple attempts to load the same
+                //       catalog, the LCIDs are filtered to retain only unique ones.
+                _Insert_lcid_if_unique(_Settings.preferred_lcid());
```

Comment:
Since `_Myelems` is guaranteed to be empty when inserting the first element, there's no need to search for duplicates of `_Lcid` in `_Myelems`.